apps/extension/src/routes/popup/internal-msg-handler.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  isPopupRequest,
+  isTxApprovalReq,
+  PopupResponse,
+  TxApproval,
+} from '@penumbra-zone/types/src/internal-msg/popup';
+import { InternalMessageHandler } from '@penumbra-zone/types/src/internal-msg/shared';
+import { useStore } from '../../state';
+import { txApprovalSelector } from '../../state/tx-approval';
+
+export const popupMsgHandler = (
+  req: unknown,
+  _: chrome.runtime.MessageSender,
+  sendResponse: (x: unknown) => void,
+) => {
+  if (isPopupRequest(req)) {
+    try {
+      if (isTxApprovalReq(req)) handleTxApproval(req.request, sendResponse);
+      else throw new Error('Unknown popup request');
+    } catch (e) {
+      const response = {
+        type: req.type,
+        error: String(e),
+      } as PopupResponse;
+      sendResponse(response);
+    }
+    return true;
```

Comment:
```suggestion
```
Per [the docs](https://developer.chrome.com/docs/extensions/reference/api/runtime#event-onMessage), looks like there's no meaning to the return value of the function you pass to `addListener`. So it's misleading to return a boolean.

Reply:
the documentation is misleading. the truthiness of a handler return indicates to the runtime if this handler intends to asynchronously provide a response.